### ISSUE TITLE
'LIBINPUT_EVENT_GESTURE_TAP_BEGIN' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN'

### ISSUE THREAD

**Issue by steve-todorov** [linuxdeepin/dde-daemon#41]
It appears that [this commit](https://github.com/linuxdeepin/dde-daemon/commit/a2404acae861135f74fecea885c0b3e3fe9193bd) has introduced a new feature that assumes `libinput` has the following event types:

```
LIBINPUT_EVENT_GESTURE_TAP_BEGIN
LIBINPUT_EVENT_GESTURE_TAP_UPDATE
LIBINPUT_EVENT_GESTURE_TAP_END
```

In the official documentation of the latest version of [libinput (1.9.1)](https://wayland.freedesktop.org/libinput/doc/latest/group__base.html#enum-members), however, there are no such event types. This results in failed build under opensuse and breaks the released version 3.2.3:

```
[   33s] # pkg.deepin.io/dde/daemon/system/gesture
[   33s] core.c: In function 'handle_gesture_events':
[   33s] core.c:214:10: error: 'LIBINPUT_EVENT_GESTURE_TAP_BEGIN' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN'?
[   33s]      case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
[   33s]           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   33s]           LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN
[   33s] core.c:214:10: note: each undeclared identifier is reported only once for each function it appears in
[   33s] core.c:216:10: error: 'LIBINPUT_EVENT_GESTURE_TAP_END' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_TAP_BEGIN'?
[   33s]      case LIBINPUT_EVENT_GESTURE_TAP_END:
[   33s]           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   33s]           LIBINPUT_EVENT_GESTURE_TAP_BEGIN
[   33s] core.c: In function 'handle_events':
[   33s] core.c:285:14: error: 'LIBINPUT_EVENT_GESTURE_TAP_BEGIN' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN'?
[   33s]          case LIBINPUT_EVENT_GESTURE_TAP_BEGIN:
[   33s]               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   33s]               LIBINPUT_EVENT_GESTURE_SWIPE_BEGIN
[   33s] core.c:286:14: error: 'LIBINPUT_EVENT_GESTURE_TAP_UPDATE' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE'?
[   33s]          case LIBINPUT_EVENT_GESTURE_TAP_UPDATE:
[   33s]               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   33s]               LIBINPUT_EVENT_GESTURE_SWIPE_UPDATE
[   33s] core.c:287:14: error: 'LIBINPUT_EVENT_GESTURE_TAP_END' undeclared (first use in this function); did you mean 'LIBINPUT_EVENT_GESTURE_TAP_BEGIN'?
[   33s]          case LIBINPUT_EVENT_GESTURE_TAP_END:{
[   33s]               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[   33s]               LIBINPUT_EVENT_GESTURE_TAP_BEGIN
[   34s] # pkg.deepin.io/dde/daemon/gesture
[   34s] gopath/src/pkg.deepin.io/dde/daemon/gesture/built-in.go:101:23: getWmHandler().TileActiveWindow undefined (type *wm.Wm has no field or method TileActiveWindow)
[   34s] gopath/src/pkg.deepin.io/dde/daemon/gesture/built-in.go:105:23: getWmHandler().TileActiveWindow undefined (type *wm.Wm has no field or method TileActiveWindow)
[   34s] gopath/src/pkg.deepin.io/dde/daemon/gesture/built-in.go:109:23: getWmHandler().BeginToMoveActiveWindow undefined (type *wm.Wm has no field or method BeginToMoveActiveWindow)
[   34s] env GOPATH="/home/abuild/rpmbuild/BUILD/dde-daemon-3.2.3/gopath:/home/abuild/go:/usr/share/go/1.9/contrib" go build  -o out/bin/soundeffect  pkg.deepin.io/dde/daemon/bin/soundeffect
[   34s] env GOPATH="/home/abuild/rpmbuild/BUILD/dde-daemon-3.2.3/gopath:/home/abuild/go:/usr/share/go/1.9/contrib" go build  -o out/bin/dde-lockservice  pkg.deepin.io/dde/daemon/bin/dde-lockservice
[   34s] env GOPATH="/home/abuild/rpmbuild/BUILD/dde-daemon-3.2.3/gopath:/home/abuild/go:/usr/share/go/1.9/contrib" go build  -o out/bin/dde-greeter-setter  pkg.deepin.io/dde/daemon/bin/dde-greeter-setter
[   34s] make: *** [Makefile:47: out/bin/dde-system-daemon] Error 2
[   34s] make: *** Waiting for unfinished jobs....
[   43s] make: *** [Makefile:47: out/bin/dde-session-daemon] Error 2
[   48s] error: Bad exit status from /var/tmp/rpm-tmp.forNaL (%build)
```

**Comment by steve-todorov** [comment]
Apologies, it seems like this is a duplicate of #40.
